keps/sig-node/2033-rootless/README.md

```diff
@@ -0,0 +1,831 @@
+- [CVE-2019-11245](https://nvd.nist.gov/vuln/detail/CVE-2019-11245): Dockerfile USER instruction was ignored by kubelet
+- [CVE-2018-11235](https://nvd.nist.gov/vuln/detail/CVE-2018-11235): kubelet could execute an arbitrary command as the root via gitRepo volumes
+- Potential image extraction [zip-slip](https://snyk.io/research/zip-slip-vulnerability) vulnerabilities in CRI runtimes. Both containerd and CRI-O are working on implementing supports for new archive formats like zstd, imgcrypt, and stargz. Potentially these implementations have such vulnerabilities.
+- And lots of CRI/OCI vulnerabilities in the past.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Allow `kubelet` and `kube-proxy` to be executed inside user namespaces create by a non-root user. See ["Required changes to Kubernetes"](#required-changes-to-kubernetes).
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+[The Node-level UserNS KEP](https://github.com/kubernetes/enhancements/issues/127) is similar to this KEP, but out of scope for this KEP.
+
+While Node-level UserNS executes only containers inside UserNS. this KEP executes all the node components inside UserNS to mitigate vulnerabilities of all components,
+
+Node-level UserNS and this KEP do not conflict and can be stacked together. (Node-level UserNS inside Kubelet's UserNS.)
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1: Production cluster
+
+A user is scared of the past vulnerabilities of kubelet/CRI/OCI, and looking for a way to
+mitigate such potential vulnerabilities.
+
+So the user would want this KEP to be implemented.
+
+The user may face difficulties for deploying stateful workloads because block-based and NFS-based persistent volumes
+mostly do not work (see [Notes/Constraints/Caveats](#notesconstraintscaveats)), but this is not a huge deal,
+when the user can use managed object storages such as Amazon S3, or managed RDBs such as Amazon RDS for storing persistent data.
+
+If the user really needs to run an application that requires the root privileges, the user
+would create a mixed cluster composed of rootful nodes and rootless nodes, and set
```

Comment:
Most CSI drivers basically require to run as root. FUSE is not a good option, as @msau42 linked above. Without root, users are left with a handful of CSI drivers that are hard to update.